lib/TickLib.sol

```diff
@@ -267,18 +262,57 @@ library TickLib {
   }  
 
   // tick underestimates the price, and we udnerestimate outbound here, so price will be overestimated here
-  function outboundFromInbound(Tick tick, uint inboundAmt) internal pure returns (uint) {
-    return inboundAmt * 1e18/tick.priceFromTick_e18();
+  function outboundFromInbound(int logPrice, uint inboundAmt) internal pure returns (uint) {
+    return inboundAmt * 1e18/priceFromLogPrice_e18(logPrice);
   }
 
-  function outboundFromInboundUp(Tick tick, uint inboundAmt) internal pure returns (uint) {
+  function outboundFromInboundUp(int logPrice, uint inboundAmt) internal pure returns (uint) {
     uint prod = inboundAmt * 1e18;
-    uint price = tick.priceFromTick_e18();
+    uint price = priceFromLogPrice_e18(logPrice);
     return prod/price + (prod%price==0?0:1);
   }
 
 
 
+}
+
+library TickLib {
+
+  function inRange(Tick tick) internal pure returns (bool) {
+    return Tick.unwrap(tick) >= MIN_TICK && Tick.unwrap(tick) <= MAX_TICK;
+  }
+
+  function fromLogPrice(int logPrice, uint tickScale) internal pure returns (Tick) {
+    // Do not force logPrices to fir the tickScale (aka logPrice%tickScale==0)
```

Comment:
Why don't we want exact math here? Would it surprise makers that we don't?

Reply:
Yeah it's not clear. They have to specify the `tickScale` so they should be aware. And tooling can choose whether to error or allow not-exactly-matching `logPrice`s. 

The scenario I have in mind: offer makers have a default "price computation" function. They can just call it and pass it without thinking about tickScale and get approximated away if they want to. Otherwise they'd have to call `logPrice/tickScale*tickScale` in addition.

WDYT?